client/txn.go

```diff
@@ -102,14 +102,25 @@ func (txn *Txn) DebugName() string {
 // snapshot. Transactions default to serializable isolation. The
 // isolation must be set before any operations are performed on the
 // transaction.
-//
-// TODO(pmattis): This isn't tested yet but will be as part of the
-// conversion of client_test.go.
-func (txn *Txn) SetSnapshotIsolation() {
-	// TODO(pmattis): Panic if the transaction has already had
-	// operations run on it. Needs to tie into the Txn reset in case of
-	// retries.
-	txn.Proto.Isolation = proto.SNAPSHOT
+func (txn *Txn) SetSnapshotIsolation() error {
+	return txn.setIsolation(proto.SNAPSHOT)
+}
+
+// SetSerializableIsolation sets the transaction's isolation type to
+// serializable. Transactions default to serializable isolation. The isolation
+// must be set before any operations are performed on the transaction.
+func (txn *Txn) SetSerializableIsolation() error {
+	return txn.setIsolation(proto.SERIALIZABLE)
+}
+
+func (txn *Txn) setIsolation(isolation proto.IsolationType) error {
+	if txn.Proto.Isolation != isolation {
+		if len(txn.Proto.ID) > 0 {
+			return fmt.Errorf("cannot change the isolation level of a running transaction")
+		}
```

Comment:
I think you're misunderstanding this error. It exists so that you can't adjust the isolation level for a transaction after running any operations on the transaction.